tasks/eva_3417/src/main/groovy/eva3417/HashCollisionWithDifferentRSRemediation.groovy

```diff
@@ -0,0 +1,573 @@
+
+                            summary.getSveInDB().getHashedMessage(), summary.getSveInDB().getAccession(),
+                            summary.getSveInDB().getClusteredVariantAccession(), summary.getRsInDBValid(),
+                            summary.getSveInDB().getRemappedFrom(),
+
+                            summary.getSveInFile().getReferenceAllele(), summary.getSveInFile().getAlternateAllele(),
+                            summary.getSveInDB().getReferenceAllele(), summary.getSveInDB().getAlternateAllele(),
+
+                            summary.getCategory()}
+                    br.write(row.join(","))
+                    br.write("\n")
+                }
+            }
+        }
+    }
+
+    SubmittedVariantEntity getSveForHash(String hash) {
+        List<SubmittedVariantEntity> sveList = [sveClass, dbsnpSveClass]
+                .collectMany(ssClass -> dbEnv.mongoTemplate.find(query(where("_id").is(hash)), ssClass))
+                .flatten()
+        if (sveList.size() > 1) {
+            throw new RuntimeException("Multiple sve found for hash " + hash)
+        }
+        return sveList.get(0)
+    }
+
+    Class getCollectionOfSVE(String hash) {
+        List<SubmittedVariantEntity> sveList = dbEnv.mongoTemplate.find(query(where("_id").is(hash)), sveClass)
+        if (sveList != null && !sveList.isEmpty()) {
+            return sveClass
+        } else {
+            List<SubmittedVariantEntity> dbsnpSveList = dbEnv.mongoTemplate.find(query(where("_id").is(hash)), dbsnpSveClass)
+            if (dbsnpSveList != null && !dbsnpSveList.isEmpty()) {
+                return dbsnpSveClass
+            } else {
+                throw new RuntimeException("SVE with hash " + hash + " could not be found")
+            }
+        }
+    }
+
+    List<ClusteredVariantEntity> getCVEInDB(Long... accessions) {
+        Set<Long> accList = new ArrayList<>()
+        for (Long acc : accessions) {
+            accList.add(acc)
+        }
+        logger.info("getting CVE for accessions " + gson.toJson(accList))
+        List<ClusteredVariantEntity> cveInDBList = [cveClass, dbsnpCveClass]
+                .collectMany(cveClass -> dbEnv.mongoTemplate.find(query(where("accession").in(accList)), cveClass))
+                .flatten()
+
+        return cveInDBList
+    }
+
+    boolean checkIfValidRS(Long rs, String asm, List<ClusteredVariantEntity> cveList) {
+        if (rs == null) {
+            return false
+        } else {
+            Optional<ClusteredVariantEntity> opRS = cveList.stream()
+                    .filter(cve -> cve.getAccession() == rs && cve.getAssemblyAccession() == asm)
+                    .findAny()
+            if (opRS.isPresent()) {
+                return true
+            } else {
+                return false
+            }
+        }
+    }
+
+    CATEGORY getCollisionCategory(Long rsInFile, Long rsInDB, boolean rsInFileValid, boolean rsInDBValid) {
```

Comment:
Can we re-use the logic in the `CollisionSummary` rather than re-implementing it?

Reply:
updated